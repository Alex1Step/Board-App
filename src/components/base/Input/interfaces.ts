export interface IinputProps {
    type: string;
    label: string;
    value: string;
    onChange: (event: { target: HTMLInputElement | HTMLSelectElement }) => void;
    withoutSubstitution?: boolean;
}
