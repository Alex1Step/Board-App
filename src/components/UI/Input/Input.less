.inputHide {
    display: none;
}

.inputShow {
    display: block;
}

.textShow {
    display: block;
}

.textHide {
    display: none;
}