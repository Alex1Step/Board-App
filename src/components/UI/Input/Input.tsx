import React from "react"
import styles from './Input.less'

interface InputProps {
    type: string,
    label: string,
    value: string,
    onChange: (e: React.ChangeEvent<HTMLInputElement>)=>void
}

const Input = (props: InputProps) => {
    const inputType: string = props.type || "text";
    const htmlFor: string = `${inputType}-${Math.random()}`;
    return (
            <React.Fragment>
                <label htmlFor={htmlFor}>{props.label}</label>
                <input 
                    type={inputType}
                    id={htmlFor}
                    value={props.value}
                    onChange={props.onChange}
                />
            </React.Fragment>
    )
}

export default Input