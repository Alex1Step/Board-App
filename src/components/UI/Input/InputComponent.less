.input-container {
  display: flex;
  justify-content: flex-start;
  label {
    text-decoration: underline;
  }
  input,
  span {
    margin-left: 5px;
  }
  span:hover {
    cursor: pointer;
  }
  .inputHide {
    display: none;
  }

  .inputShow {
    display: block;
  }

  .textShow {
    display: block;
  }

  .textHide {
    display: none;
  }

  span {
    word-wrap: break-word;
  }
}
