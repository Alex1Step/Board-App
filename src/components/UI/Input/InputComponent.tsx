import React, { useState } from 'react';
import { Input } from 'antd';
import styles from './InputComponent.less';

interface InputProps {
    type: string;
    label: string;
    value: string;
    onChange: (event: { target: HTMLInputElement | HTMLSelectElement }) => void;
    withoutSubstitution?: boolean;
}

const InputComponent = (props: InputProps): JSX.Element => {
    const [hideShow, setHideShow] = useState(1);

    const inputType: string = props.type || 'text';
    const htmlFor = `${inputType}-${Math.random()}`;

    let clsInput = [styles.inputHide];
    let clsP = [styles.textShow];
    if (hideShow === 0) {
        clsInput = [styles.inputShow];
        clsP = [styles.textHide];
    }

    return props.withoutSubstitution ? (
        <Input type={inputType} id={htmlFor} value={props.value} onChange={props.onChange} autoFocus={true} />
    ) : (
        <div className={styles.inputContainer}>
            <label htmlFor={htmlFor}>{props.label}</label>
            <span
                className={clsP.join(' ')}
                onClick={() => {
                    setHideShow(0);
                }}
            >
                {props.value}
            </span>
            <Input
                ref={(ref) => ref?.focus()}
                className={clsInput.join(' ')}
                type={inputType}
                id={htmlFor}
                value={props.value}
                onChange={props.onChange}
                onBlur={() => setHideShow(1)}
            />
        </div>
    );
};

export default InputComponent;
