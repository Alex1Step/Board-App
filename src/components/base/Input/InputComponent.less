.inputContainer {
  display: flex;
  justify-content: flex-start;
  align-items: center;

  &.wrapLabel {
    flex-direction: column;
  }

  label {
    text-decoration: underline;
    margin: 10px 0px;
  }

  input,
  span {
    margin-left: 5px;
  }

  input,
  textarea {
    background-color: #bfffe8;
    border-radius: 5px;
  }

  span:hover {
    cursor: pointer;
  }

  .inputHide {
    display: none;
  }

  .inputShow {
    display: block;
    margin-bottom: 7px;
  }

  .textShow {
    display: block;
    word-wrap: break-word !important;
  }

  .textHide {
    display: none;
  }
}

.ifTextArea {
  flex-direction: column;
  align-items: flex-start;
}
