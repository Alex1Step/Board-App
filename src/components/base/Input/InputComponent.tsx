import React, { useState } from 'react';
import { Input } from 'antd';
import styles from './InputComponent.less';
import { IinputProps } from './interfaces';
import cn from 'classnames';
import { toUpper } from 'lodash';

const { TextArea } = Input;

const InputComponent = (props: IinputProps): JSX.Element => {
    const { withWrap, type, label, value, onChange, withoutSubstitution } = props;

    const [hideShow, setHideShow] = useState(1);

    const inputType: string = type || 'text';
    const htmlFor = `${inputType}-${Math.random()}`;

    return withoutSubstitution ? (
        <Input type={inputType} id={htmlFor} value={value} onChange={onChange} autoFocus={true} />
    ) : (
        <div
            className={cn({
                [styles.inputContainer]: toUpper,
                [styles.wrapLabel]: withWrap,
                [styles.ifTextArea]: inputType === 'textarea',
            })}
        >
            <label htmlFor={htmlFor}>{label}</label>
            <span
                className={cn({
                    [styles.textShow]: hideShow,
                    [styles.textHide]: !hideShow,
                })}
                onClick={() => {
                    setHideShow(0);
                }}
            >
                {value}
            </span>
            {inputType === 'textarea' ? (
                <TextArea
                    rows={3}
                    ref={(ref) => ref?.focus()}
                    className={cn({
                        [styles.inputHide]: hideShow,
                        [styles.inputShow]: !hideShow,
                    })}
                    id={htmlFor}
                    value={value}
                    onChange={onChange}
                    onBlur={() => setHideShow(1)}
                />
            ) : (
                <Input
                    ref={(ref) => ref?.focus()}
                    className={cn({
                        [styles.inputHide]: hideShow === 1,
                        [styles.inputShow]: hideShow === 0,
                    })}
                    type={inputType}
                    id={htmlFor}
                    value={value}
                    onChange={onChange}
                    onBlur={() => setHideShow(1)}
                />
            )}
        </div>
    );
};

export default InputComponent;
