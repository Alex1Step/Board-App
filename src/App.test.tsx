import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Input from './components/base/forTests/TestInput';
import fetch from 'node-fetch';

const asyncTestFunc = () =>
    fetch('https://jsonplaceholder.typicode.com/todos/1')
        .then((response) => response.json())
        .then((json) => json.title);

describe('testing simple component', () => {
    const input = shallow(<Input />);

    test('render root element', () => {
        expect(input.find('div').exists()).toBeTruthy();
    });

    test('render child element', () => {
        expect(input.find('span')).toHaveLength(1);
        expect(input.find('div').children()).toHaveLength(2);
    });

    test('all child has classname "child" ', () => {
        expect(
            input
                .find('div')
                .children()
                .forEach((node) => expect(node.hasClass('child')).toBeTruthy()),
        );
    });

    it('change value in span', () => {
        expect(input.find('span').text()).toBe('hello');
        input.find('span').simulate('click');
        expect(input.find('span').text()).toBe('world');
    });

    //WITH SNAPSHOTS
    it('renders correctly', () => {
        const rend = renderer.create(<Input />).toJSON();
        expect(rend).toMatchSnapshot();
    });

    it('click function work correctly', () => {
        expect(input.childAt(0).prop('onClick')()).toEqual(44);
    });

    it('mock', () => {
        const mockFn = jest.fn(input.childAt(0).prop('onClick')).mockName('mockedFunction');
        mockFn();
        expect(mockFn.mock.calls.length).toBe(1);
    });

    it('mock have been called', () => {
        const mockFn = jest.fn(input.childAt(0).prop('onClick')).mockName('mockedFunction');
        mockFn();
        expect(mockFn).toHaveBeenCalled();
    });

    it('mock returned value', () => {
        const mock = jest.fn();
        mock.mockReturnValue(42);
        mock();
        expect(mock()).toBe(42);
    });
});

describe('async request', () => {
    test('recive responce', async () => {
        await expect(asyncTestFunc()).resolves.toBe('delectus aut autem');
        // await expect(Promise.resolve('lemon')).resolves.toBe('lemon');
        // await expect(Promise.resolve('lemon')).resolves.not.toBe('octopus');
    });
});
