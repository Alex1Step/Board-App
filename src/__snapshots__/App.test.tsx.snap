// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing simple component renders correctly 1`] = `
<div>
  <span
    className="child"
    onClick={[Function]}
  >
    hello
  </span>
  <p
    className="child"
  >
    hello
  </p>
</div>
`;
